test-js/unit/pentaho/type/simple.serialization.Spec.js

```diff
@@ -158,10 +224,13 @@ define([
 
       describe("values", function() {
         describe("primitive format (omitFormatted: true)", function() {
-          it("should return the primitive value", function() {
+          it("should return the primitive value when it is not a plain object or wrap it otherwise", function() {
             var spec = value.toSpec({omitFormatted: true, includeType: false});
-
-            expect(spec).toBe(simpleSpec.value);
+            if(simpleSpec.value instanceof Object && simpleSpec.value.constructor === Object) {
```

Comment:
1. Generally, agree.
2. In a way, the test title says, almost directly, which cases do what: "when the primitive value is a plain object, it returns the wrapped value", and the case `{name: "object",   value: {foo: "bar"}}` is the only case that has a primitive value which is a plain object.
3. A solution could be to add a new property to each case info object, like `isValuePlainObject`. This would allow placing the branching outside the `it` and, instead, generate one of two distinct `it`s.
4. Another solution would be to undo/refactor the forEach construction, possibly splitting it into several groups of assertions, so that the plain object case could be handled specially
5. In yet another solution, it would be left as is (because sometimes the cure is worse than the disease!).

Suggestions anyone?